massa-network-worker/src/network_worker.rs

```diff
@@ -36,40 +38,51 @@ pub struct NetworkWorker {
     /// Our node id.
     pub(crate) self_node_id: NodeId,
     /// Listener part of the establisher.
-    listener: Listener,
+    listener: Option<Listener>,
```

Comment:
Because we take it out at the beginning of the run loop, and there is a move problem if not done this way(although maybe it can be solved differently).

Reply:
Maybe worth a comment to avoid someone creating a structure without a listener and to know that we can improve this in the future.

Reply:
Right, I've added a comment.